src/dapp/libexec/dapp/dapp-verify-contract

```diff
@@ -88,41 +92,87 @@ else
   optimized=0
 fi
 
-params=(
+if  [[ $1 = "--multiple" ]]; then
+
+  # Standard Input JSON
+
+  inputJSON=$(cat <<-EOF
+  {
+    "language": "$language",
+    "sources": "$sources",
+    "settings": {
+      "remappings": "$remappings",
+      "optimizer": "$optimizer",
+      "evmVersion": "$evmVersion"
+    }
+  }
+EOF
+  )
+
+  params=(
   "module=contract" "action=verifysourcecode"
-  "contractname=$name" "contractaddress=$address"
-  "optimizationUsed=$optimized" "runs=$runs"
+  "contractname=$contractName" "contractaddress=$address"
+  "sourcecode=$inputJSON" "codeformat=solidity-standard-json-input"
   "apikey=$ETHERSCAN_API_KEY"
-)
+  )
+
+  query=$(printf "&%s" "${params[@]}")
 
-source=$(hevm flatten --source-file "$file" --json-file "$DAPP_JSON" --dapp-root "$DAPP_ROOT")
+  count=0
+  while [ $count -lt 5 ]; do
+    sleep 10
 
-source=$(cat <<.
-// Verified using https://dapp.tools
+    response=$(curl -fsS "$ETHERSCAN_API_URL" -d "$query" \
+    --data-urlencode "compilerversion=$version" \
+    --data-urlencode "constructorArguements=$constructorArguments" -X POST)
+    # NOTE: the Arguements typo is in etherscan's API
 
-$source
+    status=$(jshon <<<"$response" -e status -u)
+    guid=$(jshon <<<"$response" -e result -u)
+    message=$(jshon <<<"$response" -e message -u)
+    count=$((count + 1))
+
+    [[ $status = 1 ]] && break;
+  done
```

Comment:
This same code is used in both branches of the if statement, can you please refactor to remove the duplication (e.g. by pulling this block out of the `if` or extracting it into a seperate function)?

Reply:
Done. Let me know if it's ok!